### PR TITLE
Rework frontend trait to use XiEvent

### PR DESCRIPTION
**Pull request by bytebuddha**

### CODE REVIEW

File: src/frontend.rs [hunk 1]
```diff
@@ -12,10 +12,23 @@ use client::Client;
...
+    Update(Update),
+    ScrollTo(ScrollTo),
+    DefStyle(Style),
+    AvailablePlugins(AvailablePlugins),
+    UpdateCmds(UpdateCmds),
+    PluginStarted(PluginStarted),
+    PluginStoped(PluginStoped),
+    ConfigChanged(ConfigChanged),
+    ThemeChanged(ThemeChanged)
+}
```
Comment (Cogitri) [inline src/frontend.rs]: I suppose the following messages aren't incorporated into this:

* "alert"
* "available_themes"
*  "find_status"
* "replace_status"
* "measure_width"
* "available_languages"
* "language_changed"

Comment (bytebuddha) [reply src/frontend.rs]: You'd be correct. This was just the one's i needed for xi-term. I might close this PR though, I've yet to have time to really test it, and i probably wont for a while. I just wanted to make sure people could see the api i eventually imagine for xrl

Comment (Cogitri) [reply src/frontend.rs]: Ah, alright. Maybe you could add the missing ones and I can test this with gxi, this seems a lot more elegant than the current API :)

Comment (bytebuddha) [reply src/frontend.rs]: I might as well merge this now those rpcs are currently unimplemented in xrl

**Pull request merged**